libpromises/mod_custom.c

```diff
@@ -635,17 +660,30 @@ static inline bool HasResultAndResultIsValid(JsonElement *response)
     return ((result != NULL) && StringEqual(result, "valid"));
 }
 
-static inline const char *LogLevelToRequestFromModule()
+static inline const char *LogLevelToRequestFromModule(const Promise *pp)
 {
+    LogLevel log_level = LogGetGlobalLevel();
+
+    /* Check if there is a log level specified for the particular promise. */
+    const char *value = PromiseGetConstraintAsRval(pp, "log_level", RVAL_TYPE_SCALAR);
+    if (value != NULL)
+    {
+        LogLevel specific = ActionAttributeLogLevelFromString(value);
```

Comment:
I guess it makes sense to use the same function as for other promise types, but FYI, this one is not that good:

```
LogLevel ActionAttributeLogLevelFromString(const char *log_level)
{
    if (!log_level)
    {
        return LOG_LEVEL_ERR;
    }

    if (strcmp("inform", log_level) == 0)
    {
        return LOG_LEVEL_INFO;
    }
    else if (strcmp("verbose", log_level) == 0)
    {
        return LOG_LEVEL_VERBOSE;
    }
    else
    {
        return LOG_LEVEL_ERR;
    }
}
```

We have another one, which handles all levels, `info` vs `inform`, etc.:

```
LogLevel LogLevelFromString(const char *const level)
{
    // Only compare the part the user typed
    // i/info/inform/information will all result in LOG_LEVEL_INFO
    size_t len = SafeStringLength(level);
    if (len == 0)
    {
        return LOG_LEVEL_NOTHING;
    }
    if (StringEqualN_IgnoreCase(level, "CRITICAL", len))
    {
        return LOG_LEVEL_CRIT;
    }
    if (StringEqualN_IgnoreCase(level, "errors", len))
    {
        return LOG_LEVEL_ERR;
    }
    if (StringEqualN_IgnoreCase(level, "warnings", len))
    {
        return LOG_LEVEL_WARNING;
    }
    if (StringEqualN_IgnoreCase(level, "notices", len))
    {
        return LOG_LEVEL_NOTICE;
    }
    if (StringEqualN_IgnoreCase(level, "information", len))
    {
        return LOG_LEVEL_INFO;
    }
    if (StringEqualN_IgnoreCase(level, "verbose", len))
    {
        return LOG_LEVEL_VERBOSE;
    }
    if (StringEqualN_IgnoreCase(level, "debug", len))
    {
        return LOG_LEVEL_DEBUG;
    }
    return LOG_LEVEL_NOTHING;
}
```

Separate issue, I suppose.

Reply:
I used the one which was used in the other place that handles the `log_level` attribute. I guess it makes sense to have this handled consistently.